### ISSUE TITLE
Ability to extend input object types in the Annotation-based approach

### ISSUE THREAD

**Issue by tobias-tengler** [ChilliCream/hotchocolate#4098]
**Is your feature request related to a problem? Please describe.**
Currently there's no way (to my knowledge) of extending input object types in the Annotation-based approach.

**Describe the solution you'd like**
There should be an additional attribute `ExtendInputObjectType` that behaves the same way `ExtendObjectType` behaves for object types.

**Describe alternatives you've considered**
We could also have `ExtendObjectType` work for input object types as well, without introducing an additional attribute.

**Comment by michaelstaib** [comment]
How would such an input work? I considered it but never got around finalizing and idea on this.

If you have two classes forming one single input, how do we inject it and transport the data... we would have to hold two separate instances for a single argument or property. When to pass on which? There are a multitude of problems attached to this.

But if you have a good idea we can explore it.

**Comment by tobias-tengler** [comment]
Ah I thought these issues were already addressed, since I saw an `InputObjectTypeExtension` type floating around in the code base. Guess I'll first explore how and if it works in the Code-first approach and then try to think of a solution for Annotation-based.
Also I just transferred the issue from Slack, so more users can comment their viewpoints, needs, etc. I started doing this recently since I noticed a lot of great ideas / missing features are voiced on Slack, but then they get lost.
